sdk/identity/azure-identity/src/main/java/com/azure/identity/AzureCliCredential.java

```diff
@@ -32,6 +32,8 @@
 
     @Override
     public Mono<AccessToken> getToken(TokenRequestContext request) {
-        return identityClient.authenticateWithAzureCli(request);
+        return identityClient.authenticateWithAzureCli(request)
+                .onErrorResume(t -> Mono.error(new CredentialUnavailableException(
```

Comment:
I believe `onErrorMap` would be the more appropriate operator to use here since this appear to be more of a mapping operation than resuming (since it just triggers the error stream again).